### PR TITLE
Throw proper error for FieldValue transforms in arrays

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Fixes a regression introduced in #3048 which changed the context.path variable from `Path|null` to `undefined|null`.

Fixes https://github.com/firebase/firebase-js-sdk/issues/3172

### CODE REVIEW

File: packages/firestore/src/api/credentials.ts [hunk 1]
```diff
@@ -214,6 +214,7 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
       // outstanding so the response is potentially for a previous user (which
       // user, we can't be sure).
       if (this.tokenCounter !== initialTokenCounter) {
+        this.forceRefresh = true;
```
Comment (dconeybe) [inline packages/firestore/src/api/credentials.ts:217]: Is this change related to the fix at hand?

Comment (schmidt-sebastian) [reply packages/firestore/src/api/credentials.ts]: No, this is meant for a different PR. Removed.
